transaction/src/manifest/compiler.rs

```diff
@@ -190,7 +190,7 @@ mod tests {
                 Instruction::PublishPackageWithOwner {
                     code: Blob(code_hash),
                     abi: Blob(abi_hash),
-                    owner_badge: NonFungibleAddress::from_str("00ed9100551d7fae91eaf413e50a3c5a59f8b96af9f1297890a8f45c200721031b84c5567b126440995d3ed5aaba0565d71e1834604819ff9c17f5e9d5dd078f").unwrap(),
+                    owner_badge: NonFungibleAddress::from_str("0000000000000000000000000000000000000000000000000000045c200721031b84c5567b126440995d3ed5aaba0565d71e1834604819ff9c17f5e9d5dd078f").unwrap(),
```

Comment:
Better to construct directly, like `NonFungibleAddress("resource_xxx", Bytes("xxx"))`?

Reply:
Just to add to Yulong's point - please can we remove `from_str` from NonFungibleAddress completely?

Having these methods encourages people to think of NonFungibleAddress / NonFungibleId as sbor hex, which I'm very keen we move away from.

(If people want to SBOR encode them, they can scrypto_encode / scrypto_decode themselves)